src/data/equiv/mul_add.lean

```diff
@@ -130,7 +126,9 @@ not_congr (map_eq_one_iff h)
 
 end mul_equiv_class
 
-end mul_equiv_class
+@[to_additive] instance [has_mul α] [has_mul β] [mul_equiv_class F α β] : has_coe_t F (α ≃* β) :=
+⟨λ f, { to_fun := f, inv_fun := equiv_like.inv f, left_inv := equiv_like.left_inv f,
+  right_inv := equiv_like.right_inv f, map_mul' := map_mul f }⟩
```

Comment:
Are we now in danger of having two different coercions from a ring_equiv to a mul_equiv? They're probably defeq, but I don't know if lean will work that out.

Reply:
I'd even suggest removing those and if something breaks, re-add them as a specialization of this coercion.

Reply:
Where is this coercion? I mean to remove those.

Reply:
I think there might be other opened cans of worms here to consider - for instance, presumably `ring_equiv` now has a direct coercion to `equiv`, which means that we can no longer rely on the coercion being factored through `add_equiv` or `mul_equiv`.

Reply:
That the coercion doesn't factor through seems more desirable than that it does factor, since preferring either way of factoring seems pretty fragile and arbitrary. Also since everything builds successfully, apparently nothing relied on factoring before.

Reply:
One reason nothing relied on the coercion factoring through is by design: You only coerce strong concrete homs to weaker classes, henceso you basically never coerce twice.